Document how to protect routes in Quarkus

stefnotch opened the pull request:
Fix #7240

This adds a short section to the [Keycloak tutorial](https://quarkus.io/guides/security-keycloak-authorization) about optionally protecting routes with the Quarkus `application.properties` file. It mostly just links to other excellent Quarkus tutorials.

gsmet commented on docs/src/main/asciidoc/security-keycloak-authorization.adoc:
```diff
@@ -169,6 +169,12 @@ quarkus.keycloak.policy-enforcer.enable=true
+
+In the default configuration, Keycloak is responsible for managing the roles and deciding who can access which routes. 
+
+To configure the protected routes using the `application.properties file`, check the link:security-openid-connect-web-authentication[Protecting Web Applications Using OpenID Connect] tutorial. For more details, check the link:security[Security Guide].
```
```suggestion
To configure the protected routes using the `application.properties` file, check the link:security-openid-connect-web-authentication[Protecting Web Applications Using OpenID Connect] guide. For more details, check the link:security[Security guide].
```

gsmet reviewed (dismissed):
Thanks!

I added a small suggestion, can you adjust and amend your commit? Please squash everything into one commit!

stefnotch commented:
Ok, I adjusted the commit and squashed it into one.

sberyozkin commented on docs/src/main/asciidoc/security-keycloak-authorization.adoc:
```diff
@@ -169,6 +169,12 @@ quarkus.keycloak.policy-enforcer.enable=true
+In the default configuration, Keycloak is responsible for managing the roles and deciding who can access which routes. 
+
+To configure the protected routes using the `application.properties` file, check the link:security-openid-connect-web-authentication[Protecting Web Applications Using OpenID Connect] guide. For more details, check the link:security[Security guide].
+
```
Hi, FYI, `quarkus-keycloak-authorization` is not activated for the `web-app` applications which is what this section refers to. Can you please clarify what do you mean when you say `the protected routes` ?

stefnotch replied:
`the protected routes` refers to the URIs such as this one https://github.com/quarkusio/quarkus-quickstarts/blob/1fcd29b58ba3a80bf25ce3f5dab8d02ad0a56226/security-keycloak-authorization-quickstart/config/quarkus-realm.json#L477

sberyozkin replied:
@stefnotch Thanks. I'm sorry but I don't understand the new document addition. `quarkus-keycloak-authorization` is to support the OIDC `service` applications only when (as per your PR, Keycloak does the policy-driven authorization). If one has an OIDC `web-app` application which your PR links to then this `quarkus-keycloak-authorization` will not be activated.
This is why I'm confused about the message of this PR. Perhaps I'm missing the point :-), please clarify. CC @pedroigor 

Thanks

sberyozkin replied:
@stefnotch Perhaps you meant to link to `security-openid-connect` ?

stefnotch replied:
I see, I indeed missed `quarkus.oidc.application-type=web-app`. Thank you for catching it, I'll fix it.

sberyozkin replied:
@stefnotch Thank you :-)

sberyozkin reviewed (approved):
Hi @gsmet @stefnotch has fixed the link and the addition now looks good indeed

sberyozkin commented on docs/src/main/asciidoc/security-keycloak-authorization.adoc:
```diff
@@ -169,6 +169,12 @@ quarkus.keycloak.policy-enforcer.enable=true
 
 NOTE: By default, applications using the `quarkus-oidc` extension are marked as a `service` type application (see `quarkus.oidc.application-type`). This extension currently supports only such `service` type applications.
 
+=== Configuring protected routes
```
Hi @stefnotch I've had another look at this doc snippet, sorry, it is just a few lines, but I have one more proposal on how to change it. At the moment it feels it is out of context as it is placed in a section where the quickstart steps are described, showing how to protect the routes using this extension. Another minor bit is that this extension is also configured in `application.properties` :-)

IMHO it will read better if you add a single line pointing to the alternative approach at the very end of the introduction section at the top of the document:

```
To configure the protected routes using the `@RolesAllowed` annotation or Quarkus HTTP security policies, check the link:security-openid-connect[Using OpenID Connect Adapter to Protect JAX-RS Applications] guide and the link:security#authorization-of-web-endpoints-using-configuration[Security guide].
```

This way the quickstart solution section will be just about this quickstart.

Thanks

stefnotch replied:
I see, it is indeed a bit out of place. I wanted to put it in its own section, to make it very visible.

Would putting it in its own section at the end, also be a reasonable option? Maybe right after `Mapping Protected Resources`, since it's basically a recommendation to look at those tutorials after the quickstart?

sberyozkin replied:
@stefnotch, sure, how would you name this dedicated section though ? `Next Steps` ? May be `Additional Information` ? This last one might be fine, what do you think ?

stefnotch replied:
An alternative name, in line with the heading above it, would be `Configuring Protected Resources`. But `Additional Information` also seems fine.

sberyozkin replied:
Hi @stefnotch, but this extension is also about protecting the resources with the relevant configuration. That said, lets settle around your `Configuring Protected Resources` but highlight somehow this section is about the alternatives, `More About Configuring Protected Resources` or `Other Options for Configuring Protected Resources` or `Configuring Protected Resources without Keycloak`, etc, find the line you like please :-).

Please also review the proposed amendment to the section text itself but tweak as needed

Thanks, and sorry it takes us awhile to finalize this PR :-)

CC @pedroigor

stefnotch replied:
Ok, I updated it according to your suggestions.

stefnotch committed:
Document how to protect routes in Quarkus

Fix #7240
Update docs/src/main/asciidoc/security-keycloak-authorization.adoc

Co-Authored-By: Guillaume Smet <guillaume.smet@gmail.com>
aa


STuff
